docs/extending/extending_client_side.md

```diff
@@ -0,0 +1,528 @@
+-   [Rollup external](https://rollupjs.org/configuration-options/#external) and [Rollup output globals](https://rollupjs.org/configuration-options/#output-globals)
+-   [Webpack externals](https://webpack.js.org/configuration/externals/)
+-   [Parcel aliases](https://parceljs.org/features/dependency-resolution/#aliases)
+
+### Using an existing Stimulus controller in HTML
+
+```{warning}
+While many Stimulus controllers are in use, this doesn't imply stable and documented usage is supported for all.
+```
+
+The Stimulus approach aligns with this philosophy by relying on data attributes that can be often set when declaring Django Widgets and Wagtail Panels in Python or the HTML templates.
+For commonly used Controllers there may already be a Django widget or `FieldPanel` approach that supports your requirements already.
+
+Any built-in admin Stimulus Controller can be used via the Stimulus data attributes. These attributes can be declared in the HTML template for the content being used.
+All Wagtail's Controllers have identifiers that are prefixed with `w-`.
+
+#### Examples of using the `w-progress` Controller
+
+The `w-progress` Controller is used to help the user avoid clicking the same button multiple times when there may be some delayed behaviour required. Custom usage of this is supported for usage within the admin interface.
+
+-   `<button ... data-w-progress-duration-value="500" ...>` - custom duration can be declared on the element
+-   `<button ... class="custom-button" data-w-progress-active-class="custom-button--busy" ...>` - custom 'active' class to replace the default `button-longrunning-active` (must be a single string without spaces)
+-   `<button ... ><strong data-w-progress-target="label">{% trans 'Create' %}</strong></button>` - any element can be the button label (not just `em`)
+-   `<button ... data-action="w-progress#activate focus->w-progress#activate" ...>` - any event can be used to trigger the in-progress behaviour
+-   `<button ... data-action="w-progress#activate:once" ...>` - only trigger the progress behaviour once
+-   `<button ... data-action="readystatechange@document->w-progress#activate:once" data-w-progress-duration-value="5000" disabled ...>` - disabled on load (once JS starts) and becomes enabled after 5s duration
+
+### Debugging & error handling
+
+For simple debugging, you can enable the [Stimulus debug mode](https://stimulus.hotwired.dev/handbook/installing#debugging).
+
+```javascript
+window.Stimulus.debug = true;
+```
+
+You can use also the built-in [Stimulus error callback](https://stimulus.hotwired.dev/handbook/installing#error-handling) for more robust error handling.
+
+```javascript
+window.onerror = console.error;
+```
+
+### Testing your Controllers
+
+Wagtail uses an out-of-the-box Stimulus application instance, you should be able to write your tests with your package.json installation of Stimulus without issue.
```

Comment:
I don’t understand what "out-of-the-box" means here?

Reply:
Removing this, devs can work this out themselves, but agree this line was a bit confusing.